representation/add-logical-knobs.metta

```diff
@@ -0,0 +1,37 @@
+;;;;;;;; addLogicalKnobs ;;;;;;;;;
+;; Creates and adds logical subtree knobs in to a multimap
+;; Params:
+;;   $exemplar: Reference tree containing the target node.
+;;   (mkNodeId $targetId): ID of the target node in the exemplar.
+;;   $addIfInExemplar: If true, include knobs even if in exemplar.
+;;   $map: a multimap to add the new knobs into
+;; Return: a multimap of ($knobSpec $knob) and the updated tree
+
+;; helper function to the addLogicalKnobs
+;; takes a tuple of knobs and return a tuple of knob and knobSpec pairs
+(: mkPair (-> Expression Expression))
```

Comment:
Can you rename this function? It sounds like it's a type constructor for a datastructure called `Pair` which I was building :)